### ISSUE TITLE
support multiple MUD configs

### ISSUE THREAD

**Issue by holic** [latticexyz/mud#2217]
it would be nice if a single project could have multiple MUD configs, to separate things like namespaces, etc. and world modules could make use of this right now for better encapsulation

may be blocked on https://github.com/latticexyz/mud/issues/596

**Comment by qbzzt** [comment]
This is mostly already done for the client (see https://github.com/latticexyz/mud/blob/be89546a60f91f02939f2e6ba5be5f62113dd936/docs/pages/guides/extending-world/add-table.mdx#L159). In the codegen, the main change we'll need would be not to erase the existing definitions when we do codegen.

**Comment by holic** [comment]
MUD config now supports multiple namespaces: https://github.com/latticexyz/mud/pull/2968

We decided not to support multiple configs in a single project for now. Instead, each project is defined by the existence of a MUD config.
